search-service/src/main/kotlin/com/egm/stellio/search/service/AttributeInstanceService.kt

```diff
@@ -121,23 +123,48 @@ class AttributeInstanceService(
             .fetch()
             .all()
             .map {
-                rowToAttributeInstanceResult(it, temporalEntityAttribute)
+                rowToAttributeInstanceResult(it, temporalEntityAttribute, temporalQuery, withTemporalValues)
             }
             .collectList()
     }
 
     private fun rowToAttributeInstanceResult(
         row: Map<String, Any>,
-        temporalEntityAttribute: TemporalEntityAttribute
+        temporalEntityAttribute: TemporalEntityAttribute,
+        temporalQuery: TemporalQuery,
+        withTemporalValues: Boolean
     ): AttributeInstanceResult {
-        return AttributeInstanceResult(
-            attributeName = temporalEntityAttribute.attributeName,
-            instanceId = row["instance_id"]?.let { (it as String).toUri() },
-            datasetId = temporalEntityAttribute.datasetId,
-            value = row["value"]!!,
-            observedAt =
-                row["time_bucket"]?.let { ZonedDateTime.parse(it.toString()).toInstant().atZone(ZoneOffset.UTC) }
-                    ?: row["observed_at"].let { ZonedDateTime.parse(it.toString()).toInstant().atZone(ZoneOffset.UTC) }
-        )
+        return when {
+            withTemporalValues -> SimplifiedAttributeInstanceResult(
+                attributeName = temporalEntityAttribute.attributeName,
+                datasetId = temporalEntityAttribute.datasetId,
+                value = row["value"]!!,
+                observedAt =
+                    row["time_bucket"]?.let { ZonedDateTime.parse(it.toString()).toInstant().atZone(ZoneOffset.UTC) }
+                        ?: row["observed_at"]
+                            .let { ZonedDateTime.parse(it.toString()).toInstant().atZone(ZoneOffset.UTC) }
+            )
+            temporalQuery.timeBucket != null -> {
+                val value = row["value"]!!
+                val observedAt = row["time_bucket"]!!
+                    .let { ZonedDateTime.parse(it.toString()).toInstant().atZone(ZoneOffset.UTC) }
+                FullAttributeInstanceResult(
+                    attributeName = temporalEntityAttribute.attributeName,
+                    datasetId = temporalEntityAttribute.datasetId,
+                    value = value,
+                    observedAt = observedAt,
+                    payload = buildAttributeInstancePayload(value, observedAt, temporalEntityAttribute.datasetId)
```

Comment:
when you have a timeBucket, it should be similar to a simplified representation (so using a `SimplifiedAttributeInstanceResult` object), no?

Reply:
There is no payload in the `SimplifiedAttributeInstanceResult` model and in this case the payload is required that's why i used the `FullAttributeInstanceResult` model.

Reply:
so, to say it in other words: why is the full representation required when there is a `timeBucket`?

Reply:
because we should return a full representation of a single attribute instance that aggregates the requested instances.

Reply:
hmm, no ...